### PR TITLE
29427 models to sql

### PR DESCRIPTION
**Pull request by simon-tarr**
This branch **name** is now out of date - this branch doesn't attempt to convert django model data to an example SQL output.

Instead, this branch is an update to the underlying Django models which facilitate the many:many relationships between source table:field pairs and OMOP table:fields.

The rationale for the relationship of many:many is;
1) Each table:field pair in a source data set can potentially be associated with multiple fields in OMOP
2) Each unique pair of OMOP tables:fields can be associated with many different source table:field pairs

I thought many:many might be a useful structure for in the future. For example, we may want to go back and check where each OMOP field has been mapped (back to the original source data) for integrity checks. I don't think this would be possible if we had a 1:many relationship between the source table and the mapping table. Would be happy to discuss ideas if you like.

Example data can be loaded into a test SQLite db. Instructions on how to do this can be found in the ReadMe.

### CODE REVIEW

File: mapping_mvp/initial_data/tables_app_data.json [hunk 1]
```diff
@@ -1,37 +1 @@
...
..."dataset": "foobar", "table": "patient", "field": "gender", "mapping": [10, 11]}}, {"model": "tables.source", "pk": 4, "fields": {"dataset": "TRUST B", "table": "observation", "field": "id", "mapping": [7, 10, 11]}}, {"model": "tables.mapping", "pk": 1, "fields": {"table": "person", "field": "gender_concept_id"}}, {"model": "tables.mapping", "pk": 3, "fields": {"table": "person", "field": "person_id"}}, {"model": "tables.mapping", "pk": 4, "fields": {"table": "person", "field": "year_of_birth"}}, {"model": "tables.mapping", "pk": 5, "fields": {"table": "person", "field": "month_of_birth"}}, {"model": "tables.mapping", "pk": 6, "fields": {"table": "person", "field": "day_of_birth"}}, {"model": "tables.mapping", "pk": 7, "fields": {"table": "observation_period", "field": "observation_period_id"}}, {"model": "tables.mapping", "pk": 8, "fields": {"table": "observation_period", "field": "person_id"}}, {"model": "tables.mapping", "pk": 9, "fields": {"table": "observation_period", "field": "period_type_concept_id"}}, {"model": "tables.mapping", "pk": 10, "fields": {"table": "foo", "field": "bar"}}, {"model": "tables.mapping", "pk": 11, "fields": {"table": "bar", "field": "foobarrrrrrr"}}]
```
Comment (py5gol) [inline mapping_mvp/initial_data/tables_app_data.json:1]: Format json for readability.

File: mapping_mvp/tables/models.py [hunk 2]
```diff
@@ -1,29 +1,67 @@
...
-# Create model which defines the OMOP mapping values
-class OmopMapping(models.Model):
-    controlled_vocab_id = models.ForeignKey(ControlledVocab, on_delete=models.CASCADE)
-    omop_table = models.CharField(max_length = 265, unique = False)
-    omop_column = models.CharField(max_length = 265, unique = False)
-    omop_value = models.CharField(max_length = 265, unique = False)
 
-    def __str__(self):
-        return f'{self.omop_table, self.omop_column} - {self.omop_value}'
+
```
Comment (py5gol) [inline mapping_mvp/tables/models.py:26]: Please remove empty lines and code. Please run 'black' against the code before committing.

**Review by py5gol — CHANGES_REQUESTED** [review]

**Comment by simon-tarr** [comment]
@py5gol Addressed all your changes in your review.

File: mapping_mvp/initial_data/tables_app_data.json [hunk 3]
```diff
@@ -1 +1,135 @@
...
+[
```
Comment (simon-tarr) [inline mapping_mvp/initial_data/tables_app_data.json:1]: I don't know why Github is showing a load of whitespace at the top - if you open the file, there's no whitespace.

Comment (simon-tarr) [reply mapping_mvp/initial_data/tables_app_data.json:1]: Or is Github simply showing whitespace where old data was lurking?

**Review by py5gol — APPROVED** [review]

**Pull request merged**